src/server/services/procedures/wildcam/wildcam.js

```diff
@@ -0,0 +1,139 @@
+    return res;
+}
+function strictParseDate(val) {
+    let match = val.match(/^(\d{4})-([01]?\d)-([0-3]?\d)$/);
+    if (match) return new Date(parseInt(match[1]), parseInt(match[2]) - 1, parseInt(match[3]));
+
+    match = val.match(/^([01]?\d)\/([0-3]?\d)\/(\d{4})$/);
+    if (match) return new Date(parseInt(match[3]), parseInt(match[1]) - 1, parseInt(match[2]));
+
+    throw Error(`unknown date format '${val}'`);
+}
+
+const DATA = (function() {
+    const csvLines = fs.readFileSync(CSV_PATH, 'utf8').split(/\r?\n/);
+    const res = {};
+
+    for (let i = 1; i < csvLines.length; ++i) {
+        try {
+            const items = csvLines[i]
+                .split(',')
+                .map(x => x.trim())
+                .map(x => x.length && x[0] === x[x.length - 1] && x[0] === '"' ? x.substring(1, x.length - 1) : x);
+
+            if (items.length === 0 || (items.length === 1 && items[0] === '')) continue;
+            if (items.length !== 24) throw Error(`wrong col count - expected 24 got ${items.length}`);
+
+            let entry = res[items[23]];
+            if (!entry) entry = res[items[23]] = {
+                date: strictParseDate(items[4]),
+                timePeriod: items[8].split(' ')[0].toLowerCase(),
+                latitude: strictParseFloat(items[3]),
+                longitude: strictParseFloat(items[2]),
+                vegetationType: items[9].toLowerCase(),
+                waterType: items[12].toLowerCase(),
+                waterDistance: strictParseFloat(items[13]),
+                structureType: items[10].toLowerCase(),
+                structureDistance: strictParseFloat(items[11]),
+                species: [],
+                imageUrl: items[23],
+            }
+
+            entry.species.push({
+                name: items[14].toLowerCase(),
+                adults: strictParseInt(items[15]),
+                young: strictParseInt(items[21] || '0'),
+            });
+        } catch (ex) {
+            throw new Error(`line ${i+1}: ${ex}`);
+        }
+    }
+
+    const sorted = Object.values(res);
+    sorted.sort((a, b) => +a.date - +b.date);
+
+    return sorted;
+})();
+
+/**
+ * Searches the database for wildlife camera entries.
+ * Each return value includes information about the contents of its associated image.
+ * You can pass an entry to :func:`Wildcam.getImage` to get the actual image being described.
+ * 
+ * @param {Date=} startDate The earliest date to include in the results. If omitted, no starting cutoff is used for filtering.
+ * @param {Date=} stopDate The latest date to include in the results. If omitted, no stopping cutoff is used for filtering.
+ * @returns {Array<Object>} all data entries matching the search, in chronological order
+ */
+Wildcam.search = function (startDate = null, stopDate = null) {
+    startDate = startDate ? +startDate : -Infinity;
+    stopDate = stopDate ? +stopDate : Infinity;
+
+    return DATA.filter(x => {
```

Comment:
In other services, I have run into performance issues using just the built in methods like this when the data is sorted (or some other similar special case). If performance is a problem here and `DATA` is already sorted, it might be worth taking an iterator-like approach where you can use `skipWhile` followed by `takeWhile` to prevent it from needlessly checking items following the point where the stop date has been exceeded. (Technically, it would be faster to use binary search to find the start rather than a naive `skipWhile` over an iterator.)